### ISSUE TITLE
False positive with preview feature `companion block` (EmptyClassBlock/VariableNaming)

### ISSUE THREAD

**Issue by MukjepScarlet** [detekt/detekt#9744]
code:

```kotlin
data class A(
  val s: String
) {
  companion {
    @JvmField
    val Empty = A("")
  }
}
```

## Expected Behavior

No warn

## Observed Behavior

`EmptyClassBlock` + `VariableNaming`

## Steps to Reproduce

Previous code

## Context

KT >= 2.4.20
`freeCompilerArgs.add("-Xcompanion-blocks-and-extensions")`

## Your Environment

* Version of detekt used: 2.0.0-alpha.6
* Version of Gradle used (if applicable): 
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository):

**Comment by MukjepScarlet** [comment]
Also affects `const val`
